docs/Getting Started/Integration Basics/index.md

```diff
@@ -1236,3 +1236,201 @@ The `SDLSessionBean` should be inside a Java package other than the default pack
+    .setLanguageDesired(SDL.rpc.enums.Language.EN_US)
+    .setAppTypes([
+        SDL.rpc.enums.AppHMIType.DEFAULT,
+    ]);
+```
+
+#### Transport Configuration
+The transport configuration will depend on the environment you're using. For instance, you can make a TCP connection with NodeJS, but not in the browser. Setting up the transport correctly will be covered in the next section. See below for example transport configurations.
+
+```js
+// Example configurations to be completed by the developer
+
+// NodeJS TCP
+lifecycleConfig.setTransportConfig(new SDL.transport.TcpClientConfig(HOST, PORT));
+
+// NodeJS WebSocket Server
+lifecycleConfig.setTransportConfig(
+    new SDL.transport.WebSocketServerConfig(
+        wsClient, // The developer is responsible for providing incoming client connections
+        CONNECTION_LOST_TIMEOUT // connection timeout in milliseconds
+    )
+);
+
+// Plain JS WebSocket Client
+lifecycleConfig.setTransportConfig(new SDL.transport.WebSocketClientConfig(HOST, PORT));
+```
+
+##### App Icon
+An app icon can be set in the `LifecycleConfig` to automatically upload and set the icon image. Note that although the implementation of retreiving files are different between the JS browser and NodeJS environments, the developer can use the same API in both cases, and the SDL library will cover the implementation details for the developer depending on which build they are using.
+
+```js
+const filePath = './app_icon.png';
+const file = new SDL.manager.file.filetypes.SdlFile()
+    .setName('AppIcon')
+    .setFilePath(filePath)
+    .setType(SDL.rpc.enums.FileType.GRAPHIC_PNG)
+    .setPersistent(true);
+
+lifecycleConfig.setAppIcon(file);
+```
+
+In this case, the code snippet expects there to be an `app_icon.png` file present in the same directory for the app icon.
+
+##### SDL Manager
+
+After creating the `LifecycleConfig`, it can be set into the `AppConfig` and then passed into the `SdlManager`. The following snippet will set up the `SdlManager` and start it up. A listener is attached to the manager listener to know when the managers are ready. Listeners can also be set up for RPCs in `SdlManager`. In the snippet below, the `OnHMIStatus` notification is listened on to know what permissive state the app is in.
+
+```js
+const appConfig = new SDL.manager.AppConfig()
+    .setLifecycleConfig(lifecycleConfig);
+
+const managerListener = new SDL.manager.SdlManagerListener()
+    .setOnStart((sdlManager) => {
+        // managers are ready
+    })
+    .setOnError((sdlManager, info) => {
+        console.error('Error from SdlManagerListener: ', info);
+    });
+
+const sdlManager = new SDL.manager.SdlManager(appConfig, managerListener)
+    .start()
+    .addRpcListener(SDL.rpc.enums.FunctionID.OnHMIStatus, (onHmiStatus) => {
+        // HMI Level updates
+        const hmiLevel = onHmiStatus.getHmiLevel();
+        console.log("Current HMI Level: ", hmiLevel);
+    });
+```
+
+##### WebEngine Configuration
```

Comment:
```suggestion
## WebEngine App Configuration
```

Why is this here below setting up the config and SDL manager? Shouldn't this be up where setting up a vanilla JS vs. Node.js app is?

Reply:
Parts will be reorganized and moved to the Installation section